diff --git a/configure.py b/configure.py
index 35fc985..b797853 100644
--- a/configure.py
+++ b/configure.py
@@ -376,6 +376,8 @@ def create_makefiles(macros):
     if opts.use_qmake:
         sipconfig.inform("Creating sip code generator .pro file...")
 
+        if not os.path.isdir("sipgen"):
+            os.makedirs("sipgen")
         pro = open(os.path.join("sipgen", "sipgen.pro"), "w")
 
         pro.write("TEMPLATE = app\n")
@@ -406,7 +407,7 @@ def create_makefiles(macros):
             # There will only be one element.
             files, path = gen_installs[0]
             pro.write("\n")
-            pro.write("sip_h.files = %s\n" % " ".join(files))
+            pro.write("sip_h.files = %s\n" % files)
             pro.write("sip_h.path = %s\n" % quote(path))
             pro.write("INSTALLS += sip_h\n")
 
@@ -444,7 +445,7 @@ def create_makefiles(macros):
     all_installs.extend(module_installs)
 
     if not opts.no_module:
-        if sys.platform == 'win32':
+        if 'win' in build_platform:
             mod = 'sip.lib' if opts.static else 'sip.pyd'
         else:
             mod = 'libsip.a' if opts.static else 'sip.so'
@@ -454,6 +455,8 @@ def create_makefiles(macros):
     if opts.use_qmake:
         sipconfig.inform("Creating sip module .pro file...")
 
+        if not os.path.isdir("siplib"):
+            os.makedirs("siplib")
         pro = open(os.path.join("siplib", "siplib.pro"), "w")
 
         pro.write("TEMPLATE = lib\n")
@@ -483,7 +485,7 @@ def create_makefiles(macros):
         if not opts.static:
             # These only need to be correct for Windows.
             debug_suffix = "_d" if opts.debug else ""
-            link_lib_dir = quote("-L" + cfg.py_lib_dir)
+            link_lib_dir = quote("-L" + cfg.py_lib_dir) + " " + quote("-lpython38")
 
             pro.write("""
 win32 {
